src/main/java/com/koczy/kurek/mizera/thesisbrowser/schedule/TasksWorker.java

```diff
@@ -57,4 +58,30 @@ public void updateCitationNumber() {
         log.info("Citation numbers updated");
     }
 
+    @Scheduled(fixedRate = DAY)
+    public void updateRelatedTheses() {
+        log.info("Started updating related theses");
+        int numOfTheses = thesisDao.getNumTheses();
+        int currentThesisNumber = initialThesisNumber;
+
+        while (currentThesisNumber < numOfTheses) {
+            Thesis currentThesis = thesisDao.getNthThesis(currentThesisNumber);
+            if (Objects.nonNull(currentThesis.getTitle())) {
+                Object[] authors = currentThesis.getAuthors().toArray();
+                if (authors.length > 0) {
+                    Author firstAuthor = (Author) authors[0];
+                    List<String> relatedTheses = googleScholarScraper.getRelatedTheses(firstAuthor.getName(),
+                            currentThesis.getTitle());
+                    if (relatedTheses.size() > currentThesis.getRelatedTheses().size()) {
+                        currentThesis.setRelatedTheses(relatedTheses);
+                        thesisDao.saveThesis(currentThesis);
+                    }
+                }
+            }
+            currentThesisNumber += NEXT_THESIS_NUM;
+        }
+        initialThesisNumber = this.cal.get(Calendar.DAY_OF_MONTH) % NEXT_THESIS_NUM;
```

Comment:
Czy to nie powinno być na górze tej metody? Nie będzie teraz tak ze wykona się dla 0 i dopiero następnego dnia dla tej liczby ?